### PR TITLE
added api/timesheet DELETE

### PR DESCRIPTION
**Pull request by raduvatav**
## Description
added the possibility to DELETE timesheet entries via API

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

** I'll do the checklist below if needed **

## Checklist
- [x] I have executed `bin/console kimai:phpcs --fix` to make sure my changes adopt the correct code style
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] I agree that this code is used in Kimai and will be published under the [MIT license](https://github.com/kevinpapst/kimai2/blob/master/LICENSE)

### CODE REVIEW

**Comment by raduvatav** [comment]
I couldn't run the tests because I couldn't install phpunit... `coposer install` doesn't do that. I have `APP_ENV=dev` in .env.

Please instruct what to do to be able to test locally.

Thanks!

File: src/API/TimesheetController.php [hunk 1]
```diff
@@ -307,4 +307,36 @@ public function patchAction(Request $request, string $id)
...
+        }
+        if (!$this->isGranted('delete', $timesheet)) {
+            throw $this->createAccessDeniedException('User cannot delete timesheet');
+        }
+
+        $entityManager = $this->getDoctrine()->getManager();
+        $entityManager->remove($timesheet);
+        $entityManager->flush();
+
+        $view = new View($timesheet, 204);	// OK, No Content
```
Comment (kevinpapst) [inline src/API/TimesheetController.php]: Don't return the $timesheet. Better would be:
```php
view = new View(null, Response::HTTP_NO_CONTENT);
```

File: src/API/TimesheetController.php [hunk 2]
```diff
@@ -307,4 +307,36 @@ public function patchAction(Request $request, string $id)
...
+        $view->getContext()->setGroups(['Default', 'Entity', 'Timesheet']);
```
Comment (kevinpapst) [inline src/API/TimesheetController.php]: remove that line

File: tests/API/TimesheetControllerTest.php [hunk 3]
```diff
@@ -379,6 +379,45 @@ public function testInvalidPatchAction()
...
+            'end' => (new \DateTime())->format('Y-m-d H:m'),
+            'description' => 'foo',
+            'fixedRate' => 2016,
+            'hourlyRate' => 127
+        ];
+        $this->request($client, "/api/timesheets/$id", 'DELETE', [], json_encode($data));
+        $this->assertTrue($client->getResponse()->isSuccessful());
+
+        // check whether id is removed
+        $this->assertAccessDenied($client, "/api/timesheets/$id");
```
Comment (kevinpapst) [inline tests/API/TimesheetControllerTest.php]: Please use `$this->assertEntityNotFound(User::ROLE_USER, sprintf('/api/timesheets/:s', $id));` instead

File: tests/API/TimesheetControllerTest.php [hunk 4]
```diff
@@ -379,6 +379,45 @@ public function testInvalidPatchAction()
...
+        $data = [
+            'activity' => 1,
+            'project' => 1,
+            'begin' => (new \DateTime('- 8 hours'))->format('Y-m-d H:m'),
+            'end' => (new \DateTime())->format('Y-m-d H:m'),
+            'description' => 'foo',
+            'fixedRate' => 2016,
+            'hourlyRate' => 127
+        ];
+        $this->request($client, "/api/timesheets/$id", 'DELETE', [], json_encode($data));
```
Comment (kevinpapst) [inline tests/API/TimesheetControllerTest.php]: I am pretty sure that DELETE does not need to get the entity content. So remove `json_encode($data)` here.

File: tests/API/TimesheetControllerTest.php [hunk 5]
```diff
@@ -379,6 +379,45 @@ public function testInvalidPatchAction()
...
+        $this->assertTrue($client->getResponse()->isSuccessful());
+
+        $result = json_decode($client->getResponse()->getContent(), true);
+        $this->assertIsArray($result);
+        $this->assertDefaultStructure($result);
+        $this->assertNotEmpty($result['id']);
+        $id = $result['id'];
+
+        // now delete it
+        $data = [
```
Comment (kevinpapst) [inline tests/API/TimesheetControllerTest.php]: You can delete this, it is not needed

**Comment by horlabs** [comment]
> Regarding phpunit: it should be installed when executing `composer install`. You don't have `bin/phpunit`?

That doesn't work for me either, i had to download it manually (https://phpunit.de/getting-started/phpunit-7.html)

**Comment by kevinpapst** [comment]
Thanks for your suggestion about the new function.
I added it to master, so closing this one in favor of #726

**Pull request closed without merging**